src/SearchInput/SearchInput.tsx

```diff
@@ -150,59 +166,147 @@ export const SearchInput = forwardRef<HTMLInputElement, SearchInputProps>(
       }
     }
 
+    const handleClick = () => {
+      if (searchQuery !== null && 2 <= searchQuery.length) {
+        updateSearchQuery(searchQuery)
+        setShowOptions(true)
+      } else if (selectedValue !== null && searchQuery === null) {
+        setSearchQuery(
+          filteredList.find((val) => val.label === selectedValue)?.label ??
+            null,
+        )
+        setShowOptions(true)
+      }
+    }
+
     const handleInputChange = (event: ChangeEvent<HTMLInputElement>): void => {
       const nextValue = event.currentTarget.value
       updateSearchQuery(nextValue)
     }
 
     const handleSelect = (nextValue: SearchInputItem): void => {
       updateSearchQuery(null)
-
       setSelectedValue(nextValue.label)
       onFound(nextValue.value)
     }
 
+    const handleClearSearch = () => {
+      updateSearchQuery(null)
+      setSelectedValue(null)
```

Comment:
Should we also be clearing the selected value if the user types further in the `SearchInput`?

I'm guessing we don't want a customer to submit a form with a selected value that is different to what's showing up in the `SearchInput`

I've recorded a weird interaction that happens when you manually clear the input after selecting a value:

https://github.com/marshmallow-insurance/smores-react/assets/63399235/b5a9bd29-1bb2-495a-ba57-1b9b524a3f20

Reply:
yeah it’s a bit funky but I think this is the current behaviour right now - it does fix itself once you delete and then begin typing again

Reply:
Are we losing intended functionality if we clear `selectedValue` when a user starts typing again?

I'm opposed to use a component like this on AP with this current functionality. I wouldn't want agents to unknowingly submit values.

A similiar edgecase that I haven't highlighted in my recording is when we don't completely clear the input e.g. it could be 'Aba' - user goes to submit thinking it's going to submit 'Aba' but it submits 'Abattoir'

Reply:
that seems to be the current behaviour of the search component right now

im open to a solution but think its worth doing as its own change - as im already introducing functionality here and dont want to alter the existing behaviour much more

Reply:
thinking more about this, i think you're right and this should probably be done as this work - as we are now auto-closing the SearchOptions here, implying the value _could_ be okay

where as before, we would not close the SearchOptions until there was a valid value from the list inside

Reply:
Will refactor this